### ISSUE TITLE
🐛 lint/a11y/useButtonType doesn't know when type is spread

### ISSUE THREAD

**Issue by hornta** [biomejs/biome#2248]
### Environment information

```block
CLI:
  Version:                      1.6.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v18.18.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.15.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

```ts
const props = { type: "submit" as const };
return <button {...props}>button</button>;
```
Biome should be able to tell that the type prop is spread onto the button element but it is still reported as an error in the `lint/a11y/useButtonType` rule.

### Expected result

It should not raise any error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Sec-ant** [comment]
I think this is an expected limitation because of a lack of type inference as of now.

**Comment by ematipico** [comment]
Yeah, that's a bug. If there's a spread prop, we shouldn't trigger the rule

**Comment by fireairforce** [comment]
i want to work on this~
